scripts/prepare-open-api-for-sdk/utils.ts

```diff
@@ -0,0 +1,125 @@
+import colors from "colors";
+import { omit } from "lodash";
+
+export const isObject = (value) => {
+  return typeof value === "object" && value !== null && !Array.isArray(value);
+};
+
+export const removeStoplightTag = (node: object): object => {
+  delete node["x-stoplight"];
+  for (const attr in node) {
+    if (isObject(node[attr])) {
+      removeStoplightTag(node[attr]);
+    }
+  }
+  return node;
+};
+
+export const parseNullToNullable = (schemas) => {
+  let counter = 0;
+  const parseNullToNullableInSchema = (schema) => {
+    if (schema instanceof Object) {
+      if (schema.type === "null") {
+        counter++;
+        return { ...schema, nullable: true, type: "object" };
+      }
+
+      if (schema.properties) {
+        for (const propertyKey of Object.keys(schema.properties)) {
+          schema.properties[propertyKey] = parseNullToNullableInSchema(
+            schema.properties[propertyKey]
+          );
+        }
+      }
+    }
+
+    return schema;
+  };
+
+  for (const schemaName of Object.keys(schemas)) {
+    schemas[schemaName] = parseNullToNullableInSchema(schemas[schemaName]);
+  }
+
+  console.log(colors.green(`Replaced ${counter} nulls to nullable`));
+
+  return schemas;
+};
+
+export const removeAdditionalProperties = (
+  e: any,
+  keepIfPropertiesNotPresent: boolean
+) => {
+  const simplifyObjectModel = (o) =>
+    omit(o, ["additionalProperties", "properties"]);
+
+  if (e instanceof Object) {
+    if (
+      !keepIfPropertiesNotPresent &&
+      "additionalProperties" in e &&
+      e.additionalProperties instanceof Object
+    ) {
+      return simplifyObjectModel(e);
+    } else if (
+      "additionalProperties" in e &&
+      e.additionalProperties instanceof Object &&
+      "properties" in e
+    ) {
+      return simplifyObjectModel(e);
+    }
+    for (const f of Object.keys(e)) {
+      if (typeof e[f] === "object") {
+        e[f] = removeAdditionalProperties(e[f], keepIfPropertiesNotPresent);
+      }
+    }
+  }
+  return e;
+};
+
```

Comment:
To jest ogólnie tak strasznie przekombinowane, że przeczytanie tego jest bardzo trudne. Proponuje jakieś prostsze rozwiązanie np.

```js
export const removeRequiredOnNullable = (e) => {
  if (e.properties) {
    Object.keys(e.properties).forEach((key) => {
      e.properties[key] = removeRequiredOnNullable(e.properties[key]);
    });

    const nullables = Object.keys(e.properties).filter((key) => {
      return e.properties[key].nullable ?? false;
    })

    const required = e.required ?? [];

    const newRequired = _.difference(required, nullables);

    if (newRequired.length > 0) {
      e.required = _.difference(required, nullables);
    }
  }

  return e;
}
```

Z tego co widzę, to chyba działa to tak jak poprzednia wersja. Ale lepiej sprawdź jeszcze czy są jakieś.

No i zamiana e na schema ^